### ISSUE TITLE
avoid passing logger around

### ISSUE THREAD

**Issue by vladak** [oracle/opengrok#2470]
https://docs.python.org/3/library/logging.html#logging.getLogger says:
> All calls to this function with a given name return the same logger instance. This means that logger instances never need to be passed between different parts of an application.

This is exactly what is being done in tools. Also we should leverage child loggers as described on https://www.electricmonk.nl/log/2017/08/06/understanding-pythons-logging-module/.

**Comment by tulinkry** [comment]
isn't this solved?

**Comment by vladak** [comment]
No, you can still see the logger being passed to init routines all over the place.
